java/ql/lib/semmle/code/java/Expr.qll

```diff
@@ -161,18 +161,58 @@ class CompileTimeConstantExpr extends Expr {
     )
   }
 
+  /**
+   * Gets the stringified value of this expression, where possible.
+   *
+   * The stringified version of a compile-time constant expression is the equivalent to
+   * the result of calling `String.valueOf(expr)` on the expression.
+   * 
+   * Note that this does not handle the following cases:
+   *
+   * - mathematical computations of type `long`.
+   */
+  pragma[nomagic]
+  string getStringifiedValue() {
+    result = getStringValue()
+    or
+    result = this.(Literal).getValue()
```

Comment:
I don't think this is right re: floating-point expressions? Also check what happens re: int literals expressed in hex for example?

Reply:
I was; it's probably fine then.

Reply:
Much appreciated @Marcono1234 for following up here. I thought what I had was right, but I really appreciate you verifying!

Reply:
I think this test does a nice job of confirming your assessment:

https://github.com/github/codeql/pull/8360/files#diff-5c5952823fbe7d5baaa07b97ee5b43627e904a86112101461d6be2e8eb6e4b06